packages/filesystem/src/node/node-filesystem.ts

```diff
@@ -329,6 +330,42 @@ export class FileSystemNode implements FileSystem {
         return this.getFileStat(FileUri.create(os.homedir()).toString());
     }
 
+    getDrives(): Promise<string[]> {
+        return new Promise<string[]>((resolve, reject) => {
+            drivelist.list((error: Error, drives: { readonly mountpoints: { readonly path: string; }[] }[]) => {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+
+                const uris = drives
+                    .map(drive => drive.mountpoints)
+                    .reduce((prev, curr) => prev.concat(curr), [])
+                    .map(mountpoint => mountpoint.path)
+                    .filter(this.filterMountpointPath.bind(this))
+                    .map(path => FileUri.create(path))
+                    .map(uri => uri.toString());
+
+                resolve(uris);
+            });
+        });
+    }
+
+    /**
+     * Filters hidden and system partitions.
```

Comment:
what is considered to be `system partitions`?
would you consider mount points under `/etc` as such?

Reply:
Please, if you have any additional remarks regarding Linux, let me know. We can hide more drives.

@simark, if you have any feedback on this, please advise.

Reply:
I'm just not sure about this. maybe it's up to the application to decide what needs to be filtered.

I'm also fine with not considering `/etc` as system by default, as long as there is an option to extend and customize. 👍

Reply:
One can customize the `FileSystemNode#filterMountpointPath` and rebind the file-system.

Reply:
Hmm, at least on Linux (and I would have guessed on any Unix-style, which would have included macOS, though I'm not a regular user) it doesn't make that much sense to display the mount points.  Since the drives are all mounted in the filesystem/namespace, you just navigate it as if it was a single tree.  So it doesn't make sense to display the physical mount points in the UI, I think.